File: micronaut/src/main/java/io/swagger/model/V3SiriEstimatedTimetableSubscriptionRequest.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.model.V3SiriSubscriptionTopic;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
import org.threeten.bp.OffsetDateTime;
import io.micronaut.core.annotation.Introspected;
import io.micronaut.validation.Validated;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * V3SiriEstimatedTimetableSubscriptionRequest
 */
@Validated
@Introspected
@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.MicronautCodegen", date = "2024-04-14T03:36:01.894111500+10:00[Australia/Sydney]")
public class V3SiriEstimatedTimetableSubscriptionRequest   {
  @JsonProperty("preview_interval")
  private String previewInterval = null;

  @JsonProperty("subscriber_ref")
  private String subscriberRef = null;

  @JsonProperty("subscription_ref")
  private String subscriptionRef = null;

  /**
   * Siri Message Format 'xml' or 'json'
   */
  public enum SiriFormatEnum {
    NUMBER_0(0),
    NUMBER_1(1);

    private Integer value;

    SiriFormatEnum(Integer value) {
      this.value = value;
    }

    @Override
    @JsonValue
    public String toString() {
      return String.valueOf(value);
    }

    @JsonCreator
    public static SiriFormatEnum fromValue(String text) {
      for (SiriFormatEnum b : SiriFormatEnum.values()) {
        if (String.valueOf(b.value).equals(text)) {
          return b;
        }
      }
      return null;
    }
  }
  @JsonProperty("siri_format")
  private SiriFormatEnum siriFormat = null;

  @JsonProperty("siri_version")
  private String siriVersion = null;

  @JsonProperty("consumer_address")
  private String consumerAddress = null;

  @JsonProperty("initial_termination_time")
  private OffsetDateTime initialTerminationTime = null;

  @JsonProperty("topics")
  @Valid
  private List<V3SiriSubscriptionTopic> topics = new ArrayList<V3SiriSubscriptionTopic>();

  public V3SiriEstimatedTimetableSubscriptionRequest previewInterval(String previewInterval) {
    this.previewInterval = previewInterval;
    return this;
  }

  /**
   * Siri Preview Interval
   * @return previewInterval
  **/
  @Schema(required = true, description = "Siri Preview Interval")
  @NotNull

  public String getPreviewInterval() {
    return previewInterval;
  }

  public void setPreviewInterval(String previewInterval) {
    this.previewInterval = previewInterval;
  }

  public V3SiriEstimatedTimetableSubscriptionRequest subscriberRef(String subscriberRef) {
    this.subscriberRef = subscriberRef;
    return this;
  }

  /**
   * Siri Subscriber Ref
   * @return subscriberRef
  **/
  @Schema(required = true, description = "Siri Subscriber Ref")
  @NotNull

  public String getSubscriberRef() {
    return subscriberRef;
  }

  public void setSubscriberRef(String subscriberRef) {
    this.subscriberRef = subscriberRef;
  }

  public V3SiriEstimatedTimetableSubscriptionRequest subscriptionRef(String subscriptionRef) {
    this.subscriptionRef = subscriptionRef;
    return this;
  }

  /**
   * Siri Subscription Ref - Unique to a Subscriber Ref
   * @return subscriptionRef
  **/
  @Schema(required = true, description = "Siri Subscription Ref - Unique to a Subscriber Ref")
  @NotNull

  public String getSubscriptionRef() {
    return subscriptionRef;
  }

  public void setSubscriptionRef(String subscriptionRef) {
    this.subscriptionRef = subscriptionRef;
  }

  public V3SiriEstimatedTimetableSubscriptionRequest siriFormat(SiriFormatEnum siriFormat) {
    this.siriFormat = siriFormat;
    return this;
  }

  /**
   * Siri Message Format 'xml' or 'json'
   * @return siriFormat
  **/
  @Schema(required = true, description = "Siri Message Format 'xml' or 'json'")
  @NotNull

  public SiriFormatEnum getSiriFormat() {
    return siriFormat;
  }

  public void setSiriFormat(SiriFormatEnum siriFormat) {
    this.siriFormat = siriFormat;
  }

  public V3SiriEstimatedTimetableSubscriptionRequest siriVersion(String siriVersion) {
    this.siriVersion = siriVersion;
    return this;
  }

  /**
   * Siri Message Version '1.3' or '2.0'
   * @return siriVersion
  **/
  @Schema(required = true, description = "Siri Message Version '1.3' or '2.0'")
  @NotNull

@Pattern(regexp="1.3|2.0")   public String getSiriVersion() {
    return siriVersion;
  }

  public void setSiriVersion(String siriVersion) {
    this.siriVersion = siriVersion;
  }

  public V3SiriEstimatedTimetableSubscriptionRequest consumerAddress(String consumerAddress) {
    this.consumerAddress = consumerAddress;
    return this;
  }

  /**
   * Siri Consumer Address - Baseline and Updates will be sent to this address
   * @return consumerAddress
  **/
  @Schema(required = true, description = "Siri Consumer Address - Baseline and Updates will be sent to this address")
  @NotNull

  public String getConsumerAddress() {
    return consumerAddress;
  }

  public void setConsumerAddress(String consumerAddress) {
    this.consumerAddress = consumerAddress;
  }

  public V3SiriEstimatedTimetableSubscriptionRequest initialTerminationTime(OffsetDateTime initialTerminationTime) {
    this.initialTerminationTime = initialTerminationTime;
    return this;
  }

  /**
   * Siri Initial Termination Time - Expiry of the subscription
   * @return initialTerminationTime
  **/
  @Schema(required = true, description = "Siri Initial Termination Time - Expiry of the subscription")
  @NotNull

  @Valid
  public OffsetDateTime getInitialTerminationTime() {
    return initialTerminationTime;
  }

  public void setInitialTerminationTime(OffsetDateTime initialTerminationTime) {
    this.initialTerminationTime = initialTerminationTime;
  }

  public V3SiriEstimatedTimetableSubscriptionRequest topics(List<V3SiriSubscriptionTopic> topics) {
    this.topics = topics;
    return this;
  }

  public V3SiriEstimatedTimetableSubscriptionRequest addTopicsItem(V3SiriSubscriptionTopic topicsItem) {
    this.topics.add(topicsItem);
    return this;
  }

  /**
   * Get topics
   * @return topics
  **/
  @Schema(required = true, description = "")
  @NotNull
  @Valid
  public List<V3SiriSubscriptionTopic> getTopics() {
    return topics;
  }

  public void setTopics(List<V3SiriSubscriptionTopic> topics) {
    this.topics = topics;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3SiriEstimatedTimetableSubscriptionRequest v3SiriEstimatedTimetableSubscriptionRequest = (V3SiriEstimatedTimetableSubscriptionRequest) o;
    return Objects.equals(this.previewInterval, v3SiriEstimatedTimetableSubscriptionRequest.previewInterval) &&
        Objects.equals(this.subscriberRef, v3SiriEstimatedTimetableSubscriptionRequest.subscriberRef) &&
        Objects.equals(this.subscriptionRef, v3SiriEstimatedTimetableSubscriptionRequest.subscriptionRef) &&
        Objects.equals(this.siriFormat, v3SiriEstimatedTimetableSubscriptionRequest.siriFormat) &&
        Objects.equals(this.siriVersion, v3SiriEstimatedTimetableSubscriptionRequest.siriVersion) &&
        Objects.equals(this.consumerAddress, v3SiriEstimatedTimetableSubscriptionRequest.consumerAddress) &&
        Objects.equals(this.initialTerminationTime, v3SiriEstimatedTimetableSubscriptionRequest.initialTerminationTime) &&
        Objects.equals(this.topics, v3SiriEstimatedTimetableSubscriptionRequest.topics);
  }

  @Override
  public int hashCode() {
    return Objects.hash(previewInterval, subscriberRef, subscriptionRef, siriFormat, siriVersion, consumerAddress, initialTerminationTime, topics);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3SiriEstimatedTimetableSubscriptionRequest {\n");
    
    sb.append("    previewInterval: ").append(toIndentedString(previewInterval)).append("\n");
    sb.append("    subscriberRef: ").append(toIndentedString(subscriberRef)).append("\n");
    sb.append("    subscriptionRef: ").append(toIndentedString(subscriptionRef)).append("\n");
    sb.append("    siriFormat: ").append(toIndentedString(siriFormat)).append("\n");
    sb.append("    siriVersion: ").append(toIndentedString(siriVersion)).append("\n");
    sb.append("    consumerAddress: ").append(toIndentedString(consumerAddress)).append("\n");
    sb.append("    initialTerminationTime: ").append(toIndentedString(initialTerminationTime)).append("\n");
    sb.append("    topics: ").append(toIndentedString(topics)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3SiriEstimatedTimetableSubscriptionRequest.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3SiriEstimatedTimetableSubscriptionRequest {
    /// <summary>
    /// Siri Preview Interval
    /// </summary>
    /// <value>Siri Preview Interval</value>
    [DataMember(Name="preview_interval", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "preview_interval")]
    public string PreviewInterval { get; set; }

    /// <summary>
    /// Siri Subscriber Ref
    /// </summary>
    /// <value>Siri Subscriber Ref</value>
    [DataMember(Name="subscriber_ref", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "subscriber_ref")]
    public string SubscriberRef { get; set; }

    /// <summary>
    /// Siri Subscription Ref - Unique to a Subscriber Ref
    /// </summary>
    /// <value>Siri Subscription Ref - Unique to a Subscriber Ref</value>
    [DataMember(Name="subscription_ref", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "subscription_ref")]
    public string SubscriptionRef { get; set; }

    /// <summary>
    /// Siri Message Format 'xml' or 'json'
    /// </summary>
    /// <value>Siri Message Format 'xml' or 'json'</value>
    [DataMember(Name="siri_format", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "siri_format")]
    public int? SiriFormat { get; set; }

    /// <summary>
    /// Siri Message Version '1.3' or '2.0'
    /// </summary>
    /// <value>Siri Message Version '1.3' or '2.0'</value>
    [DataMember(Name="siri_version", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "siri_version")]
    public string SiriVersion { get; set; }

    /// <summary>
    /// Siri Consumer Address - Baseline and Updates will be sent to this address
    /// </summary>
    /// <value>Siri Consumer Address - Baseline and Updates will be sent to this address</value>
    [DataMember(Name="consumer_address", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "consumer_address")]
    public string ConsumerAddress { get; set; }

    /// <summary>
    /// Siri Initial Termination Time - Expiry of the subscription
    /// </summary>
    /// <value>Siri Initial Termination Time - Expiry of the subscription</value>
    [DataMember(Name="initial_termination_time", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "initial_termination_time")]
    public DateTime? InitialTerminationTime { get; set; }

    /// <summary>
    /// Gets or Sets Topics
    /// </summary>
    [DataMember(Name="topics", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "topics")]
    public List<V3SiriSubscriptionTopic> Topics { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3SiriEstimatedTimetableSubscriptionRequest {\n");
      sb.Append("  PreviewInterval: ").Append(PreviewInterval).Append("\n");
      sb.Append("  SubscriberRef: ").Append(SubscriberRef).Append("\n");
      sb.Append("  SubscriptionRef: ").Append(SubscriptionRef).Append("\n");
      sb.Append("  SiriFormat: ").Append(SiriFormat).Append("\n");
      sb.Append("  SiriVersion: ").Append(SiriVersion).Append("\n");
      sb.Append("  ConsumerAddress: ").Append(ConsumerAddress).Append("\n");
      sb.Append("  InitialTerminationTime: ").Append(InitialTerminationTime).Append("\n");
      sb.Append("  Topics: ").Append(Topics).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
